### PR TITLE
Add clientId argument to IBApi.connect function

### PR DESCRIPTION
**Pull request by mfrener**
I have some auto-reconnect code in my project which will re-connect if there is no more response for some time. Sometimes (when I force connection drop), TWS does not notice the lost connection immediately and so there still is an active session for my client id ad TWS reject the re-connect (ID already exists).
Therefor I would like to use new client id on ever re-connect, but right now this is not possible w/o re-creating the whole IBApi object. So solve that, IBApi.connect now has a an (optional) clientId argument and IBApiCreationOptions.clientId has been deprecated.

I have also added some information about the "Deprecation Process" on readme. 
Basically, if we want to get rid of something we add "@depricated" to give users some time to migrate and after some more time remove it.

All changes:
- Add clientId argument id IBApi.connect function
- Deprecate IBApiCreationOptions.clientId
- Deprecate IBApi.disableUseV100Plus
- Add information about deprecation process to README.md
- Updated doc

### CODE REVIEW

**Review by ceddybi — APPROVED** [review]

**Comment by Tsopic** [comment]
@mfrener how would one acquire the clientId on the first run or even before even running the API client if that's possible?

**Comment by mfrener** [comment]
Needs to come to outside as up to now (do mix it up with reqId or orderId).
Up to you now you had you to do "new IBApi({clientId: 43})". Now you can also "new IBApi().connect(43)". Just added the possiblity to re-connect with another client, w/o actually destroying IBApi object the using the constructor again.

**Comment by Tsopic** [comment]
Ahaa, so this makes it possible for us to run multiple API clients (separate accounts) at once, and the ´id´ is set by the user. That's how I understood it.

**Comment by Tsopic** [comment]
So in order to run 2 different accounts, I still need 2 instances of API gateway running. Or am I missing something?

**Comment by mfrener** [comment]
Yes, you still need two instances.
The main reason for this change is my auto-reconnect. 
If connection drops abnormally w/o proper TCP shutdown handshake (i.e. pull the network plugin after switch, so TWS will not notice there is no more network), the client session stays on TWS stays alive until TWS has detected the dead socket. This can take up to 2min (default TCP timeout on linux). During that time TWS will not accept any connection, using same client id, because it already has a session for this client id. So you need to wait session on TWS has expired and you client id is free again, or you delete the IBApi object and re-create it with new client id, or (this is new) you call connect(++clientId), simply using another client on each connect.
